recipes/pangolin/all/conanfile.py

```diff
@@ -0,0 +1,383 @@
+        "with_wayland": False,
+        "with_x11": True,
+        "with_zstd": False,
+    }
+    options_description = {
+        "python_bindings": "Support Pangolin Interactive Console",
+        "with_dc1394": "Support DC1394 video input",
+        # "with_depthsense": "Support DepthSense video input",
+        "with_ffmpeg": "Support FFMPEG video input",
+        "with_jpeg": "Support JPEG image input",
+        "with_lz4": "Support LZ4 compression",
+        "with_openexr": "Support EXR image input",
+        "with_openni2": "Support OpenNI2 video input",
+        # "with_pleora": "Support Pleora video input",
+        "with_png": "Support PNG image input",
+        "with_raw": "Support raw images",
+        "with_realsense": "Support RealSense video input",
+        # "with_telicam": "Support TeliCam video input",
+        "with_tiff": "Support TIFF image input",
+        "with_toon": "Support TooN numerics library",
+        "with_uvc": "Support USB Video Devices input",
+        # "with_uvc_mediafoundation": "Support MediaFoundation UVC input",
+        "with_v4l": "Support Video4Linux input",
+        "with_wayland": "Support Wayland windowing system",
+        "with_x11": "Support X11 windowing system",
+        "with_zstd": "Support Zstd compression",
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "7",
+            "clang": "5",
+            "apple-clang": "10",
+            "Visual Studio": "15",
+            "msvc": "191",
+        }
+
+    def export_sources(self):
+        copy(self, "conan_deps.cmake", self.recipe_folder, os.path.join(self.export_sources_folder, "src"))
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.settings.os not in ["Linux", "FreeBSD"]:
+            del self.options.with_x11
+        if self.settings.os != "Linux":
+            del self.options.with_wayland
+            del self.options.with_v4l
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.options["ffmpeg"].avdevice = True
+        self.options["ffmpeg"].avformat = True
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("eigen/3.4.0", transitive_headers=True, transitive_libs=True)
+        self.requires("glew/2.2.0", transitive_headers=True, transitive_libs=True)
+        self.requires("opengl/system", transitive_headers=True, transitive_libs=True)
+        if self.settings.os in ["Linux", "FreeBSD"]:
+            self.requires("egl/system", transitive_headers=True, transitive_libs=True)
+        if self.options.get_safe("with_wayland"):
+            # Wayland 1.20+ is not compatible as of v0.9.1
+            self.requires("wayland/1.19.0")
+            self.requires("wayland-protocols/1.33")
```

Comment:
Since `wayland-protocols` isn't a runtime dependency, most recipes incorporate it as a `tool_requires`. [Here](https://github.com/conan-io/conan-center-index/blob/master/recipes/xkbcommon/all/conanfile.py#L85) is where `xkbcommon` incorporates it. The downside to the `tool_requires` approach is that the same version of the protocols isn't properly propagated across dependencies, which seems like something that should be consistent across projects.